contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    /**
+     * Transfers administrator rights for the Strategy to another account,
+     * revoking current admin roles and setting up the roles for the new admin.
+     *
+     * @notice Can only be called by the account with the ADMIN role.
+     *
+     * @param _newAdmin The new Strategy admin account.
+     */
+    function transferAdminRights(address _newAdmin) external onlyAdmin {
+        if (_newAdmin == address(0x0)) revert StrategyAdminCannotBe0Address();
+
+        if (_newAdmin == msg.sender)
+            revert StrategyCannotTransferAdminRightsToSelf();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newAdmin);
+        _setupRole(MANAGER_ROLE, _newAdmin);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _revokeRole(MANAGER_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Returns true since strategy is synchronous.
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return true;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return investedAssets() > 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice ETH & LQTY rewards of the strategy waiting to be claimed in the liquity stability pool are not included
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return stabilityPool.getCompoundedLUSDDeposit(address(this));
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        stabilityPool.provideToSP(balance, address(0));
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice will also claim unclaimed LQTY & ETH gains
+    /// @notice when amount > total deposited, all available funds will be withdrawn
+    function withdrawToVault(uint256 amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (amount == 0) revert StrategyAmountZero();
+        if (amount > investedAssets()) revert StrategyNotEnoughShares();
+
+        // withdraws underlying amount and claims LQTY & ETH rewards
+        stabilityPool.withdrawFromSP(amount);
+
+        uint256 lqtyRewards = lqty.balanceOf(address(this));
+        uint256 ethRewards = address(this).balance;
+        // backend to keep track of this and decide when to call reinvestRewards
+        emit StrategyRewardsClaimed(lqtyRewards, ethRewards);
+
+        // use balance instead of amount since amount could be greater than what was actually withdrawn
+        uint256 balance = underlying.balanceOf(address(this));
+        if (!underlying.transfer(vault, balance)) {
```

Comment:
I guess `safeTransfer` is not an option here?

Reply:
this is because we cannot use any library that uses a `delegateCall `in a Proxy contract

Reply:
👍